[Test] Login test added

JISaldana opened the pull request:
a test on login functionality was added

JISaldana committed:
[Test] Login test added

a test on login functionality was added

florencialecha commented on qa-automation/pom.xml:
```diff
@@ -82,6 +82,7 @@
 				<artifactId>maven-surefire-plugin</artifactId>
 				<version>2.22.1</version>
 				<configuration>
+
```
Para qué sería este espacio?

JISaldana replied:
iba a cambiar el verbosity a 1 y al final no lo hice

florencialecha commented on qa-automation/src/test/java/com/digitalbank/qaautomation/dataProviders/LoginData.java:
Chequeaste con Sofi si los usuarios y contraseñas está bien ponerlas así? No recuerdo si cuál era la forma correcta para tratar datos sensibles

florencialecha commented on qa-automation/src/test/java/com/digitalbank/qaautomation/tests/loginTests/LoginTest.java:
Ojo con este test, estás imprimiendo por consola el resultado y no estas chequeando. Cuidado con los nombres del test, tenés que especificar el nombre siguiendo las convenciones de buenas prácticas que vimos. Habría que ver si chequear contra la sidebar es la forma adecuada de manejarlo.

florencialecha reviewed (commented):
Hay que revisar varias cosas, principalmente la estructura del test y ajustar cuál es el objetivo del test. Acordate que cada caso de prueba es un test solo. También hay bastantes temas de nomenclatura para revisar, la diapo de buenas prácticas que dio Sofi es súper clara respecto a esto.

florencialecha commented:
Hay que revisar varias cosas, principalmente la estructura del test y ajustar cuál es el objetivo del test. Acordate que cada caso de prueba es un test solo. También hay bastantes temas de nomenclatura para revisar, la diapo de buenas prácticas que dio Sofi es súper clara respecto a esto.

JISaldana committed:
[CP-01] Name Changes

- Class "LoginTest" renamed to "LoginTestCP01"

- TestName renamed to "Login with valid credentials "

JISaldana committed:
[Test] Changes on LoginTestCP01

-A method that lets you know if the login was successful, was added to SideBarPage

-Element "bankLogo" was deleted because it is no longer necessary

-A method that lets you know if the login was unsuccessful, was added to the LoginPage

-The test is modified to be more accurate when displaying results in Allure

JISaldana committed:
[Test] Unused imports deleted

- import org.testng.Reporter; deleted

florencialecha commented on qa-automation/src/test/java/com/digitalbank/qaautomation/pages/LoginPage.java:
- Separar la definición del elemento "sufee-alert"  del método isLoginAlertPresent(). Básicamente el suffe-alert te tiene que quedar como un elemento de la clase loginPage que podrías utilizar en cualquier otro método.
- El step de login alert esta mal, en realidad el paso es chequear si te aparece una alerta al entrar al sitio. Acordate que el step se ejecuta antes correr el código del método y es lo que te indica qué va a hacer el método.

florencialecha commented on qa-automation/src/test/java/com/digitalbank/qaautomation/pages/SideBarPage.java:
- Lo mismo que la anterior. Separar el elemento de "navbar-brand" del método para que sea reutilizable.
- Me queda la duda si el chequeo isLogoPresent es el más apropiado. Mi razonamiento es que si vos estás testeando que te loggee bien, tu resultado esperado es que te lleve a homepage en vez de ver el logo del sidebar. @sofiambd seguro puede aclararnos esto

florencialecha commented on qa-automation/src/test/java/com/digitalbank/qaautomation/tests/loginTests/LoginTestCP01.java:
```diff
@@ -0,0 +1,47 @@
+import com.digitalbank.qaautomation.dataProviders.LoginData;
+import com.digitalbank.qaautomation.pages.LoginPage;
+
+public class LoginTestCP01 {
```
- Por lo que sé, el nombre de la clase no lleva el número de caso de prueba.

florencialecha commented on qa-automation/src/test/java/com/digitalbank/qaautomation/tests/loginTests/LoginTestCP01.java:
```diff
@@ -0,0 +1,47 @@
+        driver.quit();
+    }
+
+    @Description("This test should try to log in to the website with the specified credentials")
```
Acá aclararía que estás buscando testear con credenciales válidas y no con cualquier credencial, igual que pusiste abajo.

florencialecha commented on qa-automation/src/test/java/com/digitalbank/qaautomation/tests/loginTests/LoginTestCP01.java:
```diff
@@ -0,0 +1,47 @@
+        // Navigate to the home page of the web application being tested.
+        driver.get("http://digitalbank.upcamp.io/bank/login");
+
+        LoginPage loginPage = new LoginPage(driver);
```
Acá me suena que estás tratando de hacer 2 casos a la vez, lo que te comenté en la reu y en la primer revisión. Por lo que entiendo, tu test tiene 1 sólo resultado esperado y es que pase el test. O sea, un test sería tratar de loggearte existosamente con credenciales válidas y otro test sería tratar de loggearte con credenciales inválidas y que te muestre una alerta. 
De esa forma cubris ambos casos y sabés cuando te falló qué.

florencialecha commented on qa-automation/src/test/java/com/digitalbank/qaautomation/tests/loginTests/LoginTestCP01.java:
```diff
@@ -0,0 +1,47 @@
+
+    @Description("This test should try to log in to the website with the specified credentials")
+    @Test(testName = "Login with valid credentials ",dataProvider = "credentials", dataProviderClass = LoginData.class)
+    public void loginTest(String user, String pass) throws Exception {
```
Sigue el tema del nombre, estábamso usando la convención de should...when o algo que sea un poco más descriptivo sobre qué es lo que se está probando. En la ppt de buenas prácticas está explicado.
